iup/__init__.py

```diff
@@ -406,7 +445,7 @@ def select_columns(
         name of the NIS column for the uptake estimate (population %)
     date_col: str
         name of the NIS column for the date
-    group_cols: dict
+    group_cols: dict | None
```

Comment:
While it might be a touch confusing when inspecting `select_columns` for the first time, I think it makes a really intuitive interface for the user in the config.yaml. You just specify a nested dictionary of: `{"what_the_grouping_column_is_called" : "what_I_want_the_grouping_column_to_be_called"}`. This makes sure the user enters the same number of actual vs. desired grouping column names, whereas a list for each could be different lengths (raising errors downstream). 

The 3rd example you raise shouldn't be an issue, because this is only for grouping columns. While `{"raw_point_estimate": "estimate"}` would indeed rename a column, it would also cause errors downstream because the point estimate would be erroneously used as a grouping factor.